### PR TITLE
Upgrade ember 2.18->3.4

### PR DESCRIPTION
**Pull request by toddjordan**
I realize there was a branch for upgrading already, and I started there, but I quickly got lost in the amount of merge conflicts, as its diverged from master over time.  So I started clean and ran ember-cli-update + codemods, and patched up a few broken things in the aftermath and we are here.

I appreciate the work that was already done as it made this effort much easier to be able to look and see what was done.

### CODE REVIEW

**Comment by toddjordan** [comment]
Started with an LTS jump 2.18->3.4 . I'll do the next jump 3.4->3.8

**Comment by toddjordan** [comment]
Created a heroku review app if you want to check it out ☝️

File: MODULE_REPORT.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Module Report
```
Comment (sivakumar-kailasam) [inline MODULE_REPORT.md]: please remove this file

File: package.json [hunk 2]
```diff
@@ -37,56 +37,57 @@
...
+    "@ember/jquery": "^0.5.2",
+    "@ember/optional-features": "^0.6.3",
     "broccoli-asset-rev": "^3.0.0",
     "broccoli-funnel": "^2.0.1",
     "broccoli-merge-trees": "^2.0.0",
     "ember-a11y-testing": "^0.5.4",
     "ember-algolia": "0.0.10",
     "ember-anchor": "~0.2.0",
+    "ember-auto-import": "^1.2.21",
     "ember-browserify": "^1.2.2",
```
Comment (sivakumar-kailasam) [inline package.json]: let's remove browserify now that we no longer use it.

Comment (toddjordan) [reply package.json]: we need it because one of our addons requires it :-(

Comment (toddjordan) [reply package.json]: ember-algolia

Comment (sivakumar-kailasam) [reply package.json]: Oh okay. I'll raise pr there to remove it then

**Pull request merged**